django_prices/forms.py

```diff
@@ -60,13 +81,18 @@ def __init__(
                 [MinMoneyValidator(limit_value) for limit_value in min_values]
             )
 
-    def compress(self, data_list):
-        if data_list:
-            # Raise a validation error if any of the values is empty
-            # (it is possible if field has required=False)
-            if data_list[0] in self.empty_values:
-                raise ValidationError("Enter a valid amount of money")
-            if data_list[1] in self.empty_values:
-                raise ValidationError("Enter a valid currency code")
-            return Money(data_list[0], data_list[1])
-        return None
+    def compress(self, data_list: Optional[Tuple[Decimal, str]]) -> Optional[Money]:
+        """If field is optional, return None when there is no amount provided."""
+        if not data_list:
+            return None
+
+        amount, currency = data_list
+        if amount in self.empty_values and self.required:
+            raise ValidationError("Enter a valid amount of money")
+        if currency in self.empty_values and self.required:
+            raise ValidationError("Enter a valid currency code")
```

Comment:
Do users enter a currency code?

Reply:
I believe it's for those who use custom widgets.

Reply:
I'm asking about the wording: "Enter a valid currency code" vs "Select a valid currency" as the currency is unlikely to be something entered (typed into the field) directly.

Reply:
This occurs when somebody uses fixed currency form and submits (by example using curl) intentionally different value. 
Currency Field in multicurrency form has its own validation.